weldr/common.go

```diff
@@ -275,12 +280,46 @@ func (c Client) GetFilePath(route, path string) (fileName string, apiResponse *A
 		return
 	}
 
-	// The fileName returned is safe to write to
-	fileName, err = GetContentFilename(resp.Header.Get("content-disposition"))
-	if err != nil {
-		return
+	// Save to server provided filename under current directory
+	if len(path) == 0 {
+		// The fileName returned is safe to write to
+		fileName, err = GetContentFilename(resp.Header.Get("content-disposition"))
+		if err != nil {
+			return
+		}
+		fileName = filepath.Join(path, fileName)
+	} else {
+		// Is the path a directory that exists, or a file to save to?
+
+		// If it is an existing directory? Save under that.
+		var fi fs.FileInfo
+		fi, err = os.Stat(path)
+		if err == nil {
+			if fi.IsDir() {
+				fileName, err = GetContentFilename(resp.Header.Get("content-disposition"))
+				if err != nil {
+					return
+				}
+				fileName = filepath.Join(path, fileName)
+			} else {
+				fileName = path
+			}
+		} else {
+			if errors.Is(err, fs.ErrNotExist) {
+				// Does it look like a directory? A directory needs to exist.
+				if path[len(path)-1] == '/' {
+					err = fmt.Errorf("%s does not exist", path)
+					return
+				}
+				// Assume it is a file
+				fileName = path
+			} else {
+				// Some other error
+				return
+			}
+		}
```

Comment:
The condition flow here is a bit hard to read, but it works as described so I guess it's okay.

Reply:
Yeah, I played with doing it several ways, and this was the best I could come up with. It's a bit convoluted, but I want to make sure it stays compatible with previous behavior.